### PR TITLE
Refactor and fix styles for SharedCheckbox component

### PR DESCRIPTION
**Pull request by jagodarybacka**
### What

Compiled refactors of the Checkbox component from https://github.com/tahowallet/extension/pull/3455 and https://github.com/tahowallet/extension/pull/3488 as these are not going to be merged to `main` soon. 

- new styles
- allows checkbox to be disabled
- allows checkbox to show error message
- allows to display label on both sides of the checkbox

On this branch (main) checkbox is not used anywhere right now.

Latest build: [extension-builds-3495](https://github.com/tahowallet/extension/suites/13815587027/artifacts/766623220) (as of Fri, 23 Jun 2023 11:09:22 GMT).

### CODE REVIEW

File: ui/components/Shared/SharedCheckbox.tsx [hunk 1]
```diff
@@ -1,79 +1,157 @@
...
+        />
+        {labelPosition === "left" && labelElement}
+        <div
+          className={classNames("checkbox_box", {
+            checked,
+            disabled,
+            invalid: !disabled && invalid,
+          })}
+        />
+        {labelPosition === "right" && labelElement}
```
Comment (kkosiorowska) [inline ui/components/Shared/SharedCheckbox.tsx:83]: Can we use the `flex-direction: row-reverse` property for it? Like for [SharedButton](https://github.com/tahowallet/extension/blob/main/ui/components/Shared/SharedButton.tsx#L119). This way we will not have to separate the `labelElement`.

**Review by kkosiorowska — COMMENTED** [review]
I tested the checkbox for the warning about copying to the clipboard and ledger page.  Works good ✅

> On this branch (main) checkbox is not used anywhere right now.

This checkbox is based on an already existing component in the Ledger page. Replacement occurs in https://github.com/tahowallet/extension/pull/3488 - [changes](https://github.com/tahowallet/extension/pull/3488/files#diff-dd66fca8fb70eb75303676d7f6de31d9d9779f883fc6efb8728bed50b178f819L142-R147). I wonder if we should include this in the current PR.

**Comment by jagodarybacka** [comment]
> existing component in the Ledger page

Woah why it was missed by the vsc search feature 😓

**Review by kkosiorowska — APPROVED** [review]
✅✅✅

**Pull request merged**